source/puppet/4.2/reference/lang_exported.markdown

```diff
@@ -34,6 +34,8 @@ Purpose
 
 Exported resources allow nodes to share information with each other. This is useful when one node has information that another node needs in order to manage a resource --- the node with the information can construct and publish the resource, and the node managing the resource can collect it.
 
+**Note**: Exported resources only include node-specific data, and the catalog compiler constructs and publishes exported resources from that data. This means you can only export node-specific resources stored in the catalog. Generic data not stored in the catalog, such as an entire arbitrary file on a node's filesystem, can't be exported even if referenced by the source parameter of a file resource.
```

Comment:
This could stand to be smaller and tighter. I think the way to accomplish that is to not talk about node-specific `_____` and focus on the compiler / agent boundary. Something like, "Exported resources can only use the information that's available to the compiler; they can't use information that was never sent to the compiler, like the contents of an arbitrary file on some node."